securedrop/tests/migrations/migration_48a75abc0121.py

```diff
@@ -0,0 +1,383 @@
+    are removed.
+    """
+
+    JOURNO_NUM = 20
+    SOURCE_NUM = 20
+
+    def __init__(self, config):
+        self.config = config
+        self.app = create_app(config)
+
+    def load_data(self):
+        with self.app.app_context():
+            for _ in range(self.SOURCE_NUM):
+                self.add_source()
+
+            for _ in range(self.JOURNO_NUM):
+                self.add_journalist()
+
+            for i in range(self.SOURCE_NUM):
+                # add 1-3 messages from each source, some messages are set to downloaded
+                for _ in range(random.randint(1, 3)):
+                    self.add_message(i)
+                # add 0-2 files from each source, some files are set to downloaded
+                for _ in range(random.randint(0, 2)):
+                    self.add_file(i)
+
+            # add 30 replies from randomly-selected journalists to randomly-selected sources
+            for i in range(30):
+                selected_journo = random.randint(0, self.JOURNO_NUM)
+                selected_source = random.randint(0, self.SOURCE_NUM)
+                self.add_reply(selected_journo, selected_source)
+
+            # mark some files, messages, and replies as seen
+            sql = "SELECT * FROM submissions"
+            submissions = db.engine.execute(text(sql)).fetchall()
+            for submission in submissions:
+                if submission.filename.endswith('-doc.gz.gpg') and random.choice([0, 1]):
+                    selected_journo_id = random.randint(0, self.JOURNO_NUM)
+                    self.mark_file_as_seen(submission.id, selected_journo_id)
+                elif random.choice([0, 1]):
+                    selected_journo_id = random.randint(0, self.JOURNO_NUM)
+                    self.mark_message_as_seen(submission.id, selected_journo_id)
+
+            sql = "SELECT * FROM replies"
+            replies = db.engine.execute(text(sql)).fetchall()
+            for reply in replies:
+                if random.choice([0, 1]):
+                    selected_journo_id = random.randint(0, self.JOURNO_NUM)
+                    self.mark_reply_as_seen(reply.id, selected_journo_id)
+
+            # Mark some files, messages, and replies as seen
+            for submission in submissions:
+                if submission.filename.endswith('-doc.gz.gpg') and random.choice([0, 1]):
+                    selected_journo_id = random.randint(0, self.JOURNO_NUM)
+                    self.mark_file_as_seen(submission.id, selected_journo_id)
+                elif random.choice([0, 1]):
+                    selected_journo_id = random.randint(0, self.JOURNO_NUM)
+                    self.mark_message_as_seen(submission.id, selected_journo_id)
+            for reply in replies:
+                if random.choice([0, 1]):
+                    selected_journo_id = random.randint(0, self.JOURNO_NUM)
+                    self.mark_reply_as_seen(reply.id, selected_journo_id)
+
+    def check_downgrade(self):
```

Comment:
You're not using anything added in `load_data`, are you? If not, you could save the time spent populating the tables you're about to delete.

Reply:
the idea is to load a bunch of data with seen records and then downgrade to make sure that it goes smoothly even when there are seen records

Reply:
I don't think it would matter, until we enable proper foreign key behavior (more [here](https://sqlite.org/lang_droptable.html)), at which time we'd need to do more than just drop the tables in the downgrade. But it's fine, just leave it.